tests/integration/security/pass_through_filter_chain_test.go

```diff
@@ -258,82 +238,95 @@ spec:
       mode: DISABLE`,
 					expected: []expect{
 						{
-							port:   8085,
-							schema: protocol.HTTP,
-							want:   false,
+							port: &echo.Port{ServicePort: 8085, Protocol: protocol.HTTP},
+							want: false,
 						},
 						{
-							port:   8086,
-							schema: protocol.HTTP,
-							want:   true,
+							port: &echo.Port{ServicePort: 8086, Protocol: protocol.HTTP},
+							want: true,
 						},
 						{
-							port:   8087,
-							schema: protocol.TCP,
-							want:   false,
+							port: &echo.Port{ServicePort: 8087, Protocol: protocol.TCP},
+							want: false,
 						},
 						{
-							port:   8088,
-							schema: protocol.TCP,
-							want:   true,
+							port: &echo.Port{ServicePort: 8088, Protocol: protocol.TCP},
+							want: true,
 						},
 					},
 				},
 			}
 
-			for _, cluster := range ctx.Clusters() {
-				client := apps.Naked.Match(echo.InCluster(cluster)).GetOrFail(ctx, echo.Namespace(ns.Name()))
-				destination := apps.A.Match(echo.Namespace(ns.Name())).GetOrFail(ctx, echo.InCluster(cluster))
-				from := getWorkload(client, ctx)
-				destWorkload := getWorkload(destination, ctx).Address()
-				for _, tc := range cases {
-					ctx.NewSubTest(fmt.Sprintf("In %s/%v", cluster.StableName(), tc.name)).Run(func(ctx framework.TestContext) {
-						ctx.Config().ApplyYAMLOrFail(ctx, ns.Name(), tc.config)
-						util.WaitForConfig(ctx, tc.config, ns)
+			// srcFilter finds the naked app as client.
+			findNaked := func(instances echo.Instances) echo.Instances {
+				return apps.Naked.Match(echo.Service(util.NakedSvc))
+			}
+			srcFilter := []echotest.SimpleFilter{findNaked}
+			// dstFilter finds the app a and vm as destination.
+			findPodAndVM := func(instances echo.Instances) echo.Instances {
+				a := apps.A.Match(echo.Namespace(ns.Name()))
+				vm := apps.VM.Match(echo.Namespace(ns.Name()))
+				return append(a, vm...)
+			}
+			dstFilter := []echotest.SimpleFilter{findPodAndVM}
+			for _, tc := range cases {
+				echotest.New(ctx, apps.All).
+					SetupForPair(func(ctx framework.TestContext, src, dst echo.Instances) error {
+						cfg := yml.MustApplyNamespace(t, tmpl.MustEvaluate(
+							tc.config,
+							map[string]interface{}{
```

Comment:
can we just pass the dst[0].Config.Service as the param so that you can use it simplify the template to use it directly instead of using this complex `{{ (index .dst 0).Config.Service }}` syntax?

Reply:
Yes, I have updated the template parameter to make it simpler.